--- a/player/clipboard/clipboard-mac.m
+++ b/player/clipboard/clipboard-mac.m
@@ -16,8 +16,12 @@
  */
 
 #include "clipboard.h"
+
+#if HAVE_SWIFT
 #include "osdep/mac/swift.h"
+#endif
 
+#if HAVE_SWIFT
 struct clipboard_mac_priv {
     Clipboard *clipboard;
 };
@@ -57,3 +61,18 @@
     .get_data = get_data,
     .set_data = set_data,
 };
+
+#else
+static int init(struct clipboard_ctx *cl, struct clipboard_init_params *params)
+{
+    (void)cl;
+    (void)params;
+    return CLIPBOARD_UNAVAILABLE;
+}
+
+const struct clipboard_backend clipboard_backend_mac = {
+    .name = "mac",
+    .desc = "macOS clipboard",
+    .init = init,
+};
+#endif
